modules/top-level/nixpkgs.nix

```diff
@@ -167,14 +169,9 @@ in
       example = {
         system = "x86_64-linux";
       };
-      apply =
-        value:
-        let
-          elaborated = lib.systems.elaborate value;
-        in
-        # If equivalent to `hostPlatform`, make it actually identical so that `==` can be used
-        # See https://github.com/NixOS/nixpkgs/issues/278001
-        if lib.systems.equals elaborated cfg.hostPlatform then cfg.hostPlatform else elaborated;
+      # FIXME: An elaborated platform is not supported,
+      # but an `apply` function is probably still needed.
+      # See https://github.com/NixOS/nixpkgs/pull/376988
```

Comment:
Note: this will likely need some follow up work and/or assertions, but let's wait to see what the final impl looks like upstream.

Some discussion is on https://github.com/NixOS/nixpkgs/pull/376988